manifold/src/manifold.cpp

```diff
@@ -286,17 +285,11 @@ MeshRelation Manifold::GetMeshRelation() const {
  * future reference.
  */
 std::vector<int> Manifold::GetMeshIDs() const {
-  VecDH<int> meshIDs(NumTri());
-  thrust::for_each_n(
-      thrust::device, zip(meshIDs.begin(), pImpl_->meshRelation_.triBary.begin()), NumTri(),
-      GetMeshID());
-
-  thrust::sort(thrust::device, meshIDs.begin(), meshIDs.end());
-  int n = thrust::unique(thrust::device, meshIDs.begin(), meshIDs.end()) - meshIDs.begin();
-  meshIDs.resize(n);
-
   std::vector<int> out;
-  out.insert(out.end(), meshIDs.begin(), meshIDs.end());
+  out.reserve(pImpl_->meshRelation_.originalID.size());
```

Comment:
Is this still a tight set, or could it include `meshIDs` that aren't in the `meshRelation` anymore?

Reply:
This should be tight. I use the 30th bit to indicate whether the meshID was referenced in the result of the boolean operation/compose.

Reply:
Ah, forgot to make it tight for decompose. Will fix.